constraint/pkg/externaldata/cache.go

```diff
@@ -60,15 +66,60 @@ func isValidName(name string) bool {
 }
 
 func isValidURL(url string) bool {
-	if len(url) == 0 {
-		return false
-	}
-	if !strings.HasPrefix(url, "http://") && !strings.HasPrefix(url, "https://") {
-		return false
-	}
-	return true
+	return len(url) != 0
 }
 
 func isValidTimeout(timeout int) bool {
 	return timeout >= 0
 }
+
+func isValidCABundle(provider *v1alpha1.Provider) error {
+	// verify attempts to parse the caBundle as a PEM encoded certificate
+	// to make sure it is valid before adding it to the cache
+	verify := func(caBundle string) error {
+		caPem, err := base64.StdEncoding.DecodeString(caBundle)
+		if err != nil {
+			return fmt.Errorf("failed to base64 decode caBundle: %w", err)
+		}
+
+		caDer, _ := pem.Decode(caPem)
```

Comment:
do we also want to check for error here?

Reply:
I extracted this piece of code from https://github.com/open-policy-agent/cert-controller/blob/master/pkg/rotator/rotator.go#L524-L527.